app/uk/gov/hmrc/gvmshaulierstatus/actions/AuthorisedAction.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gvmshaulierstatus.actions
+
+import com.google.inject.Inject
+import play.api.mvc.{ActionBuilder, AnyContent, Request, Results}
+import uk.gov.hmrc.internalauth.client._
+
+import scala.concurrent.Future
+
+class AuthoriseAction @Inject() (val internalAuth: BackendAuthComponents) {
```

Comment:
file name is AuthorisedAction but class name is missing `d` and 
minor: you can add `@Singleton` annotation to the class

Reply:
thank you @sureshhmrc :)